### ISSUE TITLE
no-promise-executor-return should let pass "void"

### ISSUE THREAD

**Issue by danielrentz** [eslint/eslint#13523]
**Tell us about your environment**

* **ESLint Version:** 7.5.0
* **Node Version:** 12.16.2
* **npm Version:** 6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
The rule "no-promise-executor-return" should let pass "void" returns, e.g. in arrow functions.

```js
    // fails as expected
    exports.pause = t => new Promise(resolve => setTimeout(resolve, t));

    // passes as expected
    exports.pause = t => new Promise(resolve => { setTimeout(resolve, t); });

    // should pass too
    exports.pause = t => new Promise(resolve => void setTimeout(resolve, t));
```

**What did you expect to happen?**
Last example should pass.

**What actually happened? Please include the actual, raw output from ESLint.**
Rule fails.

**Are you willing to submit a pull request to fix this bug?**
no.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

This is not a bug but an enhancement I think.  (changing the label, let me know if I am wrong )

IMO, I do think it will be helpful to have for upper scope data but few data will still lost eg `return void function(){ return 2 }` (Possible Error), may be putting this behind an option

***

On sidenote, does anyone facing issue in the eslint demo page. this rule is not [report errors](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcHJvbWlzZS1leGVjdXRvci1yZXR1cm46IFwiZXJyb3JcIiovXG5cbm5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICByZXNvbHZlKGRlZmF1bHRSZXN1bHQpO1xuICAgIHJldHVybjtcbn0pO1xuXG5uZXcgUHJvbWlzZSgoKSA9PiB7XG4gICAgcmV0dXJuIDE7XG59KTtcblxubmV3IFByb21pc2UocmVzID0+IHZvaWQgZnVuY3Rpb24oKXt9KVxuXG4vLyBmYWlscyBhcyBleHBlY3RlZFxuZXhwb3J0cy5wYXVzZSA9IHQgPT4gbmV3IFByb21pc2UocmVzb2x2ZSA9PiBzZXRUaW1lb3V0KHJlc29sdmUsIHQpKTtcblxuLy8gcGFzc2VzIGFzIGV4cGVjdGVkXG5leHBvcnRzLnBhdXNlID0gdCA9PiBuZXcgUHJvbWlzZShyZXNvbHZlID0+IHsgc2V0VGltZW91dChyZXNvbHZlLCB0KTsgfSk7XG5cbi8vIHNob3VsZCBwYXNzIHRvb1xuZXhwb3J0cy5wYXVzZSA9IHQgPT4gbmV3IFByb21pc2UocmVzb2x2ZSA9PiB2b2lkIHNldFRpbWVvdXQocmVzb2x2ZSwgdCkpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7ImluZGVudCI6Miwibm8tcHJvbWlzZS1leGVjdXRvci1yZXR1cm4iOjJ9LCJlbnYiOnt9fX0=) ? just want to confirm

**Comment by mdjermanovic** [comment]
> On sidenote, does anyone facing issue in the eslint demo page. this rule is not [report errors](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcHJvbWlzZS1leGVjdXRvci1yZXR1cm46IFwiZXJyb3JcIiovXG5cbm5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICByZXNvbHZlKGRlZmF1bHRSZXN1bHQpO1xuICAgIHJldHVybjtcbn0pO1xuXG5uZXcgUHJvbWlzZSgoKSA9PiB7XG4gICAgcmV0dXJuIDE7XG59KTtcblxubmV3IFByb21pc2UocmVzID0+IHZvaWQgZnVuY3Rpb24oKXt9KVxuXG4vLyBmYWlscyBhcyBleHBlY3RlZFxuZXhwb3J0cy5wYXVzZSA9IHQgPT4gbmV3IFByb21pc2UocmVzb2x2ZSA9PiBzZXRUaW1lb3V0KHJlc29sdmUsIHQpKTtcblxuLy8gcGFzc2VzIGFzIGV4cGVjdGVkXG5leHBvcnRzLnBhdXNlID0gdCA9PiBuZXcgUHJvbWlzZShyZXNvbHZlID0+IHsgc2V0VGltZW91dChyZXNvbHZlLCB0KTsgfSk7XG5cbi8vIHNob3VsZCBwYXNzIHRvb1xuZXhwb3J0cy5wYXVzZSA9IHQgPT4gbmV3IFByb21pc2UocmVzb2x2ZSA9PiB2b2lkIHNldFRpbWVvdXQocmVzb2x2ZSwgdCkpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7ImluZGVudCI6Miwibm8tcHJvbWlzZS1leGVjdXRvci1yZXR1cm4iOjJ9LCJlbnYiOnt9fX0=) ? just want to confirm

Try to enable `es6` in `Environments` (rule reports errors only if it's a reference to global `Promise`).

**Comment by anikethsaha** [comment]
Cool, thanks a lot
